lib/transform.js

```diff
@@ -28,23 +42,96 @@ module.exports = class Transform {
      */
     applyTransforms(baseTemplate, childTemplate, fullRendering) {
+        const [template, ignoredParts] = this._ignore(baseTemplate);
         const options = { treeAdapter };
+
         const rootNodes = fullRendering
-            ? this.parse(baseTemplate, options)
-            : this.parseFragment(baseTemplate, options);
+            ? this._parse(template, options)
+            : this._parseFragment(template, options);
+
         const slotMap =
             childTemplate && typeof childTemplate === 'string'
                 ? this._groupSlots(parse5.parseFragment(childTemplate, options))
                 : new Map();
+
         const serializerOptions = {
             treeAdapter,
             slotMap,
             handleTags: this.handleTags,
             fullRendering
         };
+
         const serializer = new CustomSerializer(rootNodes, serializerOptions);
-        return serializer.serialize();
+
+        return this._restore(ignoredParts, serializer.serialize());
+    }
+
+    /**
+     * Ignore special parts of a base template which marked with the help of special HTML comments:
+     * <!-- TailorX: Ignore during parsing START --> and <!-- TailorX: Ignore during parsing END -->
+     * before and after a content which you want to ignore during parsing to speed up transforming of a base template
+     *
+     * @example <!-- TailorX: Ignore during parsing START --><div class='example-class>Example</div><!-- TailorX: Ignore during parsing END -->
+     *
+     * @param {String} baseTemplate - Base template that contains all the necessary tags and fragments for the given page (Used by multiple pages)
+     * @returns {Array} Array consiting of a template without ignored parts as a String and ignored parts as an Object
+     */
+    _ignoreBeforeParsing(baseTemplate) {
+        const parts = baseTemplate.split(
+            /<!-- TailorX: Ignore during parsing START -->|<!-- TailorX: Ignore during parsing END -->/gm
+        );
+        const ignoredParts = {};
+
+        const joinedParts = parts.reduce((parts, part, index) => {
+            if (index % 2 === 0) {
+                return parts.concat(part);
+            }
+
+            ignoredParts[index] = part;
+
+            return parts.concat(
+                `<!-- TailorX: Ignored content during parsing #${index} -->`
+            );
+        });
+
+        return [joinedParts, ignoredParts];
     }
+
+    /**
+     * Restore base template's special parts after ignoring them by @method _ignoreBeforeParsing
+     *
+     * @param {Object} ignoredParts - Base template's ignored parts
+     * @param {Array} chunks - Array consiting of Buffers and Objects
+     * @returns {Array} Array consiting of Buffers and Objects
+     */
+    _restoreAfterParsing(ignoredParts, chunks) {
+        return chunks.map(chunk => {
+            const isBuffer = chunk instanceof Buffer;
+            const part = isBuffer
+                ? Buffer.from(chunk, 'base64').toString('utf-8')
```

Comment:
Are you sure that `Buffer.from(chunk, 'base64')` is valid here? I think if chunk is an instance of Buffer you can just call `.toString('utf-8')`